apps/backend/src/libs/helpers/check-permission.ts

```diff
@@ -0,0 +1,42 @@
+import { type FastifyRequest } from "fastify";
+
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { type APIPreHandler } from "~/libs/modules/controller/libs/types/types.js";
+import { HTTPCode, HTTPError } from "~/libs/modules/http/http.js";
+import { permissionService } from "~/modules/permissions/permissions.js";
+
+const checkPermission = (requiredPermission: string): APIPreHandler => {
+	return async (request: FastifyRequest): Promise<void> => {
+		const userId = request.user?.id;
+
+		if (!userId) {
+			throw new HTTPError({
+				message: ExceptionMessage.USER_NOT_FOUND,
+				status: HTTPCode.UNAUTHORIZED,
+			});
+		}
+
+		const permissions = await permissionService.getPermissions(userId);
```

Comment:
I think there was a misunderstanding with permissions in the app. We need the permissions service for getting and manipulating all the permissions (for example, for the permissions dropdown when we create a group). But for permissions for the current user, we need to attach them to the current user through the groups. So that UserAuthResponseDto contains groups field with permissions which we need to include using relation in users getById method. We don't need a separate call for user permissions, we will use get permissions for getting all available in db.

Reply:
For this task, we need to attach permissions to the authenticated user itself. So instead of loading them separately, you already will have groups and permissions inside authenticated user. You need to include them using relations in one query (not a separate call) in getById method of users service. We don't need service/api/slice to get all permissions in this task.

Reply:
If I understand correctly, then the UserAuthResponseDto will look like this:
```
type UserAuthResponseDto = {
	createdAt: string;
	email: string;
	id: number;
	name: string;
	groups: [
		{
			groupName: string;
			permissions: [
				{
					name: string;
					key: string;
				}
			]
		}
	]
};
```

Reply:
yes, similar to this